apis/duck/v1alpha1/condition_set.go

```diff
@@ -150,10 +151,11 @@ func (r conditionsImpl) IsHappy() bool {
 // GetCondition finds and returns the Condition that matches the ConditionType
 // previously set on Conditions.
 func (r conditionsImpl) GetCondition(t ConditionType) *Condition {
-	if r.accessor == nil {
+	if r.status == nil {
```

Comment:
Again, I'd encourage not allowing a `conditionsImpl` to have a `nil` status.

Reply:
I could add in the nil handling in the `conditionsImpl` to allow `nil.DoSomething()` like we did with `FieldError`. But it will still be a silent error unless you check for nil

Reply:
That's what I'm worried about. Someone writes some reasonable-looking code, but passes in the wrong object to SetConditions. It would be nice if something would flag that there was a problem.